sw/host/opentitanlib/src/debug/dmi.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::{Deref, DerefMut};
+use std::time::Duration;
+
+use anyhow::{bail, ensure, Result};
+use thiserror::Error;
+
+use super::openocd::OpenOcd;
+use crate::test_utils::poll::poll_until;
+
+pub mod consts {
```

Comment:
Maybe add a reference to the specification and the version that this corresponds to?